### ISSUE TITLE
[serverpod_auth] User ids should not be guessable

### ISSUE THREAD

**Issue by lukehutch** [serverpod/serverpod#554]
serverpod_auth generates consecutive user ids, starting from 1. User profile images are served at URLs that include the user id. Therefore, a hacker could exfiltrate all user profile images by simply iterating through user ids.

User ids should not be guessable. It would be much better to hash `userIdentifier` (the email address) using SHA256, and then use the first 8-ish bytes of this hash code, in some base, as the user ID.

**Comment by yahu1031** [comment]
@lukehutch Yes, The id in these tables is not preferred use for UUIDs. Please use [uuid](https://pub.dev/packages/uuid) and create a column for UUID and use it.

**Comment by lukehutch** [comment]
There's actually a `hash` column already that could be used as the user ID (although it probably needs to be truncated a bit).

**Comment by yahu1031** [comment]
Then perfect. IDK about it. Thanks for Letting me know. Please feel free to share more info if you have.

**Comment by lukehutch** [comment]
I found a Postgres function that invertably maps the Uint32 space onto a permutation of the Uint32 space, based on a 32-byte key.

https://wiki.postgresql.org/wiki/Skip32_(crypt_32_bits)

This could be used for generating apparently-random 32-bit user ids from sequential integers when user accounts are created, in order to make user ids non-guessable.

**Comment by lukehutch** [comment]
Would anyone object to me implementing this id order obfuscation method as the default for `userId`, as shown in the previous comment?

I'm guessing this should use 52 bits (and therefore needs to use `bigint` in Postgres), to fit into dart ints, while still having the requisite even number of bits.

**Comment by marcelomendoncasoares** [comment]
Closing as completed since the new auth uses UUIDs for the users.
